src/Lachain.Core/RPC/HTTP/Web3/AccountServiceWeb3.cs

```diff
@@ -39,8 +40,60 @@ public class AccountServiceWeb3 : JsonRpcService
         public string GetBalance(string address, string tag)
         {
             var addressUint160 = address.HexToUInt160();
-            var availableBalance = GetSnapshotByTag(tag)!.Balances.GetBalance(addressUint160);
-            return Web3DataFormatUtils.Web3Number(availableBalance.ToWei().ToUInt256());
+
+            if (tag == "pending")
+            {
+                // Get all transaction from pool
+                var txpool = this._transactionPool;
+                var transactions = txpool.Transactions;
+
+                List<TransactionReceipt> tx_Receipts = new List<TransactionReceipt>();
+                
+                foreach(var tx in transactions)
+                {
+                    tx_Receipts.Add(tx.Value);
+                }
+
+                // Sort on the basis of nonce
+                tx_Receipts = tx_Receipts.OrderBy(receipt => receipt, new ReceiptComparer()).ToList();
+
+                // Get current address nonce
+                var transactionRepository = _stateManager.CurrentSnapshot.Transactions;
+                var curr_nonce = transactionRepository.GetTotalTransactionCount(address.HexToBytes().ToUInt160());
+
+                // Virtually execute the txs in nonce order
+                var availableBalance = GetSnapshotByTag("latest")!.Balances.GetBalance(addressUint160);
+
+                foreach (var tx in tx_Receipts)
+                {
+                    var from = tx.Transaction.From.ToHex();
+
+                    if (address == from & curr_nonce == tx.Transaction.Nonce)
+                    {
+                        // Executing the transaction
+                        var gasp = new Money(tx.Transaction.GasPrice);
+                        var gasl = new Money(tx.Transaction.GasLimit);
+                        var txamnt = new Money(tx.Transaction.Value);
+
+                        availableBalance = availableBalance - txamnt  - (gasl * gasp);
+
+                        // Check if balance is less than 0
+                        if(availableBalance < Money.Parse("0"))
+                        {
+                            return Web3DataFormatUtils.Web3Number(Money.Parse("0").ToWei().ToUInt256());
+                        }
+                    }
+
+                    curr_nonce += 1;
```

Comment:
This nonce update should be inside the If statement. 
We should only increase the nonce if the transaction's from == address